extensions/SemanticScribunto/src/ScribuntoLuaLibrary.php

```diff
@@ -0,0 +1,365 @@
+
+		// get usable result
+		$result = $this->doPostProcessParserFunctionCallResult(
+			$parserFunctionCallResult
+		);
+
+		if ( strlen( $result ) ) {
+			// if result is a non empty string, assume an error message
+			return [ [ 1 => false, self::SMW_ERROR_FIELD => preg_replace( '/<[^>]+>/', '', $result ) ] ];
+		}
+
+		// on success, return true
+		return [ 1 => true ];
+	}
+
+	/**
+	 * This mirrors the functionality of the parser function #subobject and
+	 * makes it available to lua.
+	 *
+	 * @param string|array $arguments arguments passed from lua, string or array depending on call
+	 * @param string $subobjectId if you need to manually assign an id, do this here
+	 *
+	 * @return null|array|array[]
+	 */
+	public function subobject( $arguments, $subobjectId = null ) {
+
+		$arguments = $this->processLuaArguments( $arguments );
+
+		// parameters[0] would be the subobject id, so unshift
+		array_unshift( $arguments, null );
+
+		// if subobject id was set, put it on position 0
+		if ( !is_null( $subobjectId ) && $subobjectId ) {
+			// user deliberately set an id for this subobject
+			$arguments[0] = $subobjectId;
+
+			// we need to ksort, otherwise ParameterProcessorFactory doesn't
+			// recognize the id
+			ksort( $arguments );
+		}
+
+		// prepare subobjectParserFunction object
+		$subobjectParserFunction = $this->getLibraryFactory()->newSubobjectParserFunction(
+			$this->getEngine()->getParser()
+		);
+
+		$parserFunctionCallResult = $subobjectParserFunction->parse(
+			 $this->getLibraryFactory()->newParserParameterProcessorFrom( $arguments )
+		);
+
+		if ( strlen( $result = $this->doPostProcessParserFunctionCallResult( $parserFunctionCallResult ) ) ) {
+			// if result a non empty string, assume an error message
+			return [ [ 1 => false, self::SMW_ERROR_FIELD => preg_replace( '/<[^>]+>/', '', $result ) ] ];
+		}
+
+		// on success, return true
+		return [ 1 => true ];
+	}
+
+	/**
+	 * This takes an array and converts it so, that the result is a viable lua table.
+	 * I.e. the resulting table has its numerical indices start with 1
+	 * If `$ar` is not an array, it is simply returned
+	 * @param mixed $ar
+	 * @return mixed array
+	 */
+	private function convertArrayToLuaTable( $ar ) {
+
+		if ( is_array( $ar) ) {
+			foreach ( $ar as $key => $value ) {
+				$ar[$key] = $this->convertArrayToLuaTable( $value );
+			}
+			array_unshift( $ar, '' );
+			unset( $ar[0] );
+		}
+		return $ar;
+	}
+
+	/**
+	 * Takes a result returned from a parser function call and prepares it to be
+	 * used as parsed string.
+	 *
+	 * @since 1.0
+	 *
+	 * @param string|array $parserFunctionResult
+	 *
+	 * @return string
+	 */
+	private function doPostProcessParserFunctionCallResult( $parserFunctionResult ) {
+
+		// parser function call can return string or array
+		if ( is_array( $parserFunctionResult ) ) {
+			$result = $parserFunctionResult[0];
+			$noParse = isset( $parserFunctionResult['noparse'] ) ? $parserFunctionResult['noparse'] : true;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the literal "true" boolean value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)